cpp/CMakeLists.txt

```diff
@@ -209,11 +209,19 @@ include(cmake/thirdparty/get_fmt.cmake)
 # find spdlog
 include(cmake/thirdparty/get_spdlog.cmake)
 
+rapids_cpm_find(
```

Comment:
Could we move this `rapids_cpm_find` call into a file like `cmake/thirdparty/get_nanoarrow.cmake`, and then include it? See the examples in that directory or ask if you need help.

Reply:
Something like this:

<details><summary>get_nanoarrow.cmake</summary>

```
# =============================================================================
# Copyright (c) 2024, NVIDIA CORPORATION.
#
# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
# in compliance with the License. You may obtain a copy of the License at
#
# http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software distributed under the License
# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
# or implied. See the License for the specific language governing permissions and limitations under
# the License.
# =============================================================================

function(find_and_configure_nanoarrow)
  set(oneValueArgs VERSION FORK PINNED_TAG)
  cmake_parse_arguments(PKG "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})

  rapids_cpm_find(
    nanoarrow ${PKG_VERSION}
    BUILD_EXPORT_SET cudf-exports
    INSTALL_EXPORT_SET cudf-exports
    CPM_ARGS
    GIT_REPOSITORY https://github.com/${PKG_FORK}/arrow-nanoarrow.git
    GIT_TAG ${PKG_PINNED_TAG}
    GIT_SHALLOW TRUE
  )
endfunction()

find_and_configure_nanoarrow(
  VERSION 0.4.0 FORK apache PINNED_TAG c97720003ff863b81805bcdb9f7c91306ab6b6a8
)
```

</details>

Reply:
Note that I defined export sets in the proposal above. I think that should work instead of manually exporting?

Reply:
even with those defined export sets, if I don't manually export I get 

```
CMake Error: install(EXPORT "cudf-exports" ...) includes target "cudf" which requires target "nanoarrow" that is not in any export set.
CMake Error in CMakeLists.txt:
  export called with target "cudf" which requires target "nanoarrow" that is
  not in any export set.
```

Reply:
I've pushed something that should fix this for now. We can discuss in more depth down below since there are a few questions to be addressed about proper usage of nanoarrow I think.